### PR TITLE
Odom Topic & Frame Namespaces 

### PR DESCRIPTION
**Pull request by sp-sophia-labs**
## Description

The goal of this PR is to enable namespaces on diff_drive_controller for the odom topic and the odom frame 

## How has this been tested?

We have tested this code on create3 and turtlebot4 robots. You can find our project here: https://github.com/sp-sophia-labs/turtlebot4_multi_sim.  Reviewers may clone the project and replicate exactly what we use for testing (you can find the details in the project's README)

## Checklist

- [x] Lint code with appropriate parser
- [x] Rebase code on galactic branch

### CODE REVIEW

**Comment by bmagyar** [comment]
I'd really prefer to merge this into `master` and backport to Galactic. Is that ok with you? I know you've tested it in Galactic

**Comment by sp-sophia-labs** [comment]
@bmagyar No problem for me, I've rebased the changes on master and fixed the linting with `pre-commit`

File: diff_drive_controller/src/diff_drive_controller.cpp [hunk 1]
```diff
@@ -472,8 +472,20 @@ controller_interface::CallbackReturn DiffDriveController::on_configure(
       odometry_publisher_);
 
   auto & odometry_message = realtime_odometry_publisher_->msg_;
-  odometry_message.header.frame_id = odom_params_.odom_frame_id;
-  odometry_message.child_frame_id = odom_params_.base_frame_id;
+
+  std::string controller_namespace = std::string(node_->get_namespace());
```
Comment (bmagyar) [inline diff_drive_controller/src/diff_drive_controller.cpp:476]:
```suggestion
  std::string controller_namespace = std::string(get_node()->get_namespace());
```
I think it's something like this in the new api

**Review by bmagyar — APPROVED** [review]

**Review by bmagyar — COMMENTED** [review]
Assign namespace prefix directly to odom_params to simplify code

**Review by bmagyar — CHANGES_REQUESTED** [review]
Assign namespace prefix directly to odom_params to simplify code

**Review by bmagyar — APPROVED** [review]

**Review by destogl — APPROVED** [review]

File: diff_drive_controller/src/diff_drive_controller.cpp [hunk 2]
```diff
@@ -471,6 +471,20 @@ controller_interface::CallbackReturn DiffDriveController::on_configure(
...
+  {
+    controller_namespace = "";
+  }
+  else
+  {
+    controller_namespace = controller_namespace + "/";
+  }
+
+  odom_params_.odom_frame_id = controller_namespace + odom_params_.odom_frame_id;
+  odometry_message.child_frame_id = controller_namespace + odom_params_.base_frame_id;
```
Comment (bmagyar) [inline diff_drive_controller/src/diff_drive_controller.cpp:486]:
```suggestion
  odom_params_.base_frame_id = controller_namespace + odom_params_.base_frame_id;
```

**Pull request merged**